### DISCUSSION TITLE
How to establish from TextInput afterStateUpdated() if the input is inside a repeater

### DISCUSSION THREAD

**Discussion by bvicini** [filamentphp/filament#14792] [Help]
### Package

Form builder

### Package Version

v3.2.121

### How can we help you?

Hello, I'm implementing an autosave function that triggers whenever a form component state changes, overloading afterStateUpdated() in derived classes of TextInput and other form components, for example
```
class LaraTextInput extends TextInput
{
    use CustomLabelTrait ;

    public $mandatory = 1 ;
    public static $record ;

    public function setRecord($record)
    {
        self::$record = $record;
        return $this ;
    }

    public static function make($name): static
    {
        return parent::make($name)
            ->live()
            ->reactive()
            ->afterStateUpdated(function (LaraTextInput $component,$state) {
                self::$record->safeSave($component->statePath,$state);
            });
    }
.....
```

In the form I have:
```
\Filament\Forms\Components\Repeater::make("altre_alleanze")
                ->label("a text or another")
                ->schema([
                    App\Schooldocs\LaraTextInput::make('otherAlliance')
                    ->setRecord($record)
                    ->label('Altra alleanza'),
                ]),
```
Surely there is a better way to obtain autosave (any suggestion is appreciated!), but now my problem is when the TextInput inside the Repeater changes: I cannot find a way to store the data correctly for the repeater, but my code puts it as a separate entry in the data structure to store. Any suggestion? Thank you

**Comment by lucasff** [comment]
@bvicini looking at the source of Repeater, I thnink you can't make the `TextInput` trigger the auto-save.
You have to go to the parent repeater, and from there trigger the save for all its descendants. I think it's like that to guarantee data integrity, because only the repeater is able to know about its descendants.

Try with `getChildComponentContainers` or if you manage with `TextInput`, you can try with getParentComponent to see if it returns a Repeater and then trigger the save.

I hope I was useful for you.
